### PR TITLE
FIX: Restore text logo feature after moving to upload site settings.

### PR DESCRIPTION
**Pull request by tgxworld**
https://meta.discourse.org/t/default-text-logo-not-working/103936

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -199,9 +199,12 @@ def crawlable_meta_data(opts = nil)
     opts[:url] ||= "#{Discourse.base_url_no_prefix}#{request.fullpath}"
 
     twitter_summary_large_image_url =
-      SiteSetting.site_twitter_summary_large_image_url
+      SiteSetting.twitter_summary_large_image&.url
```
Comment (SamSaffron) [inline app/helpers/application_helper.rb]: wait so why don't we use a CDN image for this?

File: app/models/site_setting.rb [hunk 2]
```diff
@@ -177,54 +177,19 @@ def self.Upload
...
+    logo_small
+    digest_logo
+    mobile_logo
+    large_icon
+    favicon
+    apple_touch_icon
+  }.each do |setting_name|
+    define_singleton_method("site_#{setting_name}_url") do
+      upload = self.public_send(setting_name)
+      upload ? full_cdn_url(upload.url) : ''
```
Comment (SamSaffron) [inline app/models/site_setting.rb]: how often do we do this upload lookup? every time you ask for the setting?

File: db/fixtures/010_uploads.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
```
[the remark this answers is not in the document]
Comment (SamSaffron) [reply db/fixtures/010_uploads.rb]: I guess ... one concern is what will happen when we switch stores. it is certainly sophisticated and somewhat wasteful in multisite. is there any clean way around this?

File: app/models/site_setting.rb (hunk 2 above)
Comment (tgxworld) [reply app/models/site_setting.rb]: Of course not :) We only do it the first time you ask for the site setting.

File: db/fixtures/010_uploads.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (tgxworld) [reply db/fixtures/010_uploads.rb]: I might be able to make it such that the upload records points at the public directory without having to actually recreate the upload file.That way the record will always remain local as well.

[the remark this answers is not in the document]
Comment (tgxworld) [reply db/fixtures/010_uploads.rb]: @SamSaffron Good point about multisite being wasteful. I've updated the code to seed the upload defaults in the DB so we don't actually have to duplicate the upload file.

**Comment by tgxworld** [comment]
@SamSaffron I can make the site text logo work but we still need to be able to support "default" for uploads site settings.

**Comment by SamSaffron** [comment]
@tgxworld closing this for now... I am open to a feature like this but we need a far simpler implementation if we bring it back.

**Comment by tgxworld** [comment]
@SamSaffron Can you elaborate on why this implementation is not ideal or complicated? A site setting backed by uploads is always going to require an upload record in the DB so I'm not seeing anyway around the seeding of the record.

**Comment by SamSaffron** [comment]
Ok go ahead and merge but put a shortcut in for the seeding to cut down 4 queries to 1. 

```
if DB.query_single('select count(*) from uploads where id in (-1,-2,-3,-4)') != 4
   ... stuff .. 
end
```

Cuts a bunch of queries on multisite for the happy path.

**Comment by tgxworld** [comment]
Merged in https://github.com/discourse/discourse/commit/b0c8fdd7da20fc4eeb30575050360fbf47ec4c8f

**Pull request closed without merging**